am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -222,6 +225,18 @@ private boolean explicitAccessType(Set<String> userRoles) {
         return roleBasedAccessRecords.stream().collect(getMapCollector());
     }
 
+    /**
+     * Retrieves a list of {@link ResourceDefinition} that user roles have create permissions for.
+     *
+     * @param userRoles a set of roles
+     * @return a list of resource definitions
+     */
+    public List<ResourceDefinition> resourceCreationAllowedList(Set<String> userRoles) {
```

Comment:
That beautiful method deserves better name. Maybe `getResourceDefinitionsWithCreatePermission`